### PR TITLE
Improve toolchain resolution debug

### PR DESCRIPTION
**Pull request by comius**
When multiple toolchain types are resolved, skyframe functions run in parallel and the debug output is often hard to follow. Adding type of toolchain to each message makes it possible to analyse the results better.

Also after a execution transition, target changes. Added target label to the message, to be able to tell for which target toolchain was selected.

Removing initial messages "Looking for toolchain of type...", "Considering toolchain...". With additional fields displayed they become obsolete.

Condensing 'rejection explanation', i.e. which values are mismatched, into a single line.

#7713, b/126375095: there is still some duplications in ToolchainResolutionFunction:125, which I didn't remove because some tests depend on it (when specifying extra_toolchains)

b/117279695 is fixed with this

### CODE REVIEW

**Comment by comius** [comment]
cc @lberki @c-mita @oquenchil

File: src/main/java/com/google/devtools/build/lib/skyframe/SingleToolchainResolutionFunction.java [hunk 1]
```diff
@@ -210,41 +205,50 @@ private static boolean checkConstraints(
       @Nullable EventHandler eventHandler,
       ConstraintCollection toolchainConstraints,
       String platformType,
-      PlatformInfo platform) {
+      PlatformInfo platform,
+      Label toolchainTypeLabel, Label toolchainLabel) {
```
Comment (katre) [inline src/main/java/com/google/devtools/build/lib/skyframe/SingleToolchainResolutionFunction.java:209]: Please put toolchainLabel on a new line for consistency,

File: src/test/shell/bazel/toolchain_test.sh [hunk 2]
```diff
@@ -533,9 +533,8 @@ EOF
     --toolchain_resolution_debug \
     --incompatible_auto_configure_host_platform \
     //demo:use &> $TEST_log || fail "Build failed"
-  expect_log 'ToolchainResolution: Looking for toolchain of type //toolchain:test_toolchain'
-  expect_log 'ToolchainResolution:   For toolchain type //toolchain:test_toolchain, possible execution platforms and toolchains: {@local_config_platform//:host -> //:test_toolchain_impl_1}'
-  expect_log 'ToolchainResolution: Selected execution platform @local_config_platform//:host, type //toolchain:test_toolchain -> toolchain //:test_toolchain_impl_1'
+  expect_log 'ToolchainResolution:   Type //toolchain:test_toolchain: target @local_config_platform//.*: execution @local_config_platform//:host: Selected toolchain //:test_toolchain_impl_1'
```
Comment (katre) [inline src/test/shell/bazel/toolchain_test.sh:536]: Instead of "target @local_config_platform", which looks like it is the target being built, specify that this is "target platform @local_config_platform".

**Review by katre — CHANGES_REQUESTED** [review]
Does this help mitigate #7713? Also see http://b/117279695 and http://b/126375095.

File: src/test/shell/bazel/toolchain_test.sh (hunk 2 above)
Comment (comius) [reply src/test/shell/bazel/toolchain_test.sh:536]: > Does this help mitigate #7713? Also see http://b/117279695 and http://b/126375095.

#7713, b/126375095: there is still some duplications in ToolchainResolutionFunction:125, which I didn't remove because some tests depend on it (when specifying extra_toolchains)

b/117279695 is fixed with this

What would be most useful for me is to have two levels: displaying selected toolchains, displaying also why they were rejected.

File: src/main/java/com/google/devtools/build/lib/skyframe/SingleToolchainResolutionFunction.java (hunk 1 above)
Comment (comius) [reply src/main/java/com/google/devtools/build/lib/skyframe/SingleToolchainResolutionFunction.java]: Done.

**Comment by comius** [comment]
> Does this help mitigate #7713? Also see http://b/117279695 and http://b/126375095.

#7713, b/126375095: there is still some duplications in ToolchainResolutionFunction:125, which I didn't remove because some tests depend on it (when specifying extra_toolchains)

b/117279695 is fixed with this

What would be most useful for me is to have two levels: displaying selected toolchains, displaying also why they were rejected.

File: src/test/shell/bazel/toolchain_test.sh (hunk 2 above)
Comment (comius) [reply src/test/shell/bazel/toolchain_test.sh]: Done.

**Review by katre — APPROVED** [review]
